From: Xavier Claessens <xavier.claessens@collabora.com>
Date: Thu, 2 Jan 2020 21:56:10 -0500
Subject: Skip G_GNUC_(BEGIN|END)_IGNORE_DEPRECATIONS lines

For some reason, glib has to put empty line before and after each of
these lines otherwise the symbol following it is undeclared.

Origin: upstream, commit:ca42972cd3fc5420a429ae752228c0c89ec7c763
---
 gtkdoc/scan.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gtkdoc/scan.py b/gtkdoc/scan.py
index d04d4d4..7de08ad 100644
--- a/gtkdoc/scan.py
+++ b/gtkdoc/scan.py
@@ -561,6 +561,11 @@ def ScanHeaderContent(input_lines, decl_list, get_types, options):
                     logging.info('Found start of comment: %s', line.strip())
                 continue
 
+            # Skip begin/end deprecation macros.
+            m = re.search(r'^\s*G_GNUC_(BEGIN|END)_IGNORE_DEPRECATIONS', line)
+            if m:
+                continue
+
             logging.info('no decl: %s', line.strip())
 
             cm = [m.match(line) for m in CLINE_MATCHER]
